workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/EntityGameObjectLinker.cs

```diff
@@ -21,9 +21,29 @@ public EntityGameObjectLinker(World world, WorkerSystem worker)
             entityManager = world.GetExistingManager<EntityManager>();
         }
 
-        public void LinkGameObjectToEntity(GameObject gameObject, Entity entity, EntityId spatialEntityId,
-            ViewCommandBuffer viewCommandBuffer)
+        public void LinkGameObjectToEntity(GameObject gameObject, Entity entity, ViewCommandBuffer viewCommandBuffer)
         {
+            bool hasSpatialEntityId = entityManager.HasComponent<SpatialEntityId>(entity);
+            bool isWorkerEntity = entityManager.HasComponent<WorkerEntityTag>(entity);
+            if (!hasSpatialEntityId && !isWorkerEntity)
+            {
+                worker.LogDispatcher.HandleLog(LogType.Warning, new LogEvent(
+                        "Attempted to link GameObject to an Entity that is not a SpatialOS Entity or the worker entity," +
+                        "this is not possible")
```

Comment:
/giphy "inconceivable!"

I'd not say "this is not possible", the part before this is clear enough